HIP/hip-260.md

```diff
@@ -122,6 +126,68 @@ final updated value of the storage slot after the completion of the smart
 contract call. Transient states between the start and finish of the contract are
 not stored in the record stream.
 
+### Record Stream File Changes
+
+Currently Hedera nodes produce record stream files that contain a series of
+transactions within a preconfigured interval. The record stream file (.rcd) 
+contains a transaction record for each transaction in that file. For each 
+record stream file there is a corresponding signature file (rcd_sig) 
+that includes the signature generated by the node.
+
+#### Side car records file
+
+A new file of type (.rcd_side) will be introduced to track all side car records
+produced during the same preconfigured interval as the main record file (.rcd). All
+side car records in the .rcd_side file should have a corresponding main record
+with the same id in the main record file (.rcd).
+
+The sidecar + record file + signature file will all be zipped together into a 
+single zip and uploaded to the AWS/Google buckets. A lambda running on AWS/Google 
+will unzip and store the files separately. Mirror nodes will have the option as 
```

Comment:
This process doesn't sound very decentralized. Even though the bucket is owned by Hedera, only individual nodes should have permission to write to their individual paths within the bucket. This will become important once we have community and permissionless nodes. Having a lambda that can write to paths that only nodes should write to sounds centralized.

Instead, I would suggest that the record file and sidecar be gzipped on the node and uploaded separately to their final position. The signature is not worth gzipping separate as it is actually smaller if you don't zip it. This will ensure a similar amount of overall bytes is uploaded while still being decentralized.